crates/router/src/core/payments.rs

```diff
@@ -1229,6 +1232,45 @@ where
     router_data_res
 }
 
+async fn blocklist_guard<F, ApiRequest, Ctx>(
+    state: &AppState,
+    merchant_account: &domain::MerchantAccount,
+    operation: &BoxedOperation<'_, F, ApiRequest, Ctx>,
+    payment_data: &mut PaymentData<F>,
+) -> CustomResult<bool, errors::ApiErrorResponse>
+where
+    F: Send + Clone + Sync,
+    Ctx: PaymentMethodRetrieve,
+{
+    let merchant_id = &payment_data.payment_attempt.merchant_id;
+    let blocklist_enabled_key = format!("guard_blocklist_for_{merchant_id}");
+    let blocklist_guard_enabled = state
+        .store
+        .find_config_by_key_unwrap_or(&blocklist_enabled_key, Some("false".to_string()))
+        .await;
+
+    let blocklist_guard_enabled: bool = match blocklist_guard_enabled {
+        Ok(config) => serde_json::from_str(&config.config).unwrap_or(false),
+
+        // If it is not present in db we are defaulting it to false
+        Err(inner) => {
+            if !inner.current_context().is_db_not_found() {
+                logger::error!("Error fetching guard blocklist enabled config {:?}", inner);
+            }
+            false
+        }
```

Comment:
is this logic doing what you intend to do? There is a negation of the condition, and the return value is outside the if block.

Reply:
@Narayanbhat166 we already have major refactorings ongoing for the above feature, which will result in most of the code becoming depricated. So we can possibly do all refactorings in the upcoming PRs, as this is critical to go.